### ISSUE TITLE
Result types are incorrectly inferred when `undefined` explicitly passed to `select`/`include`

### ISSUE THREAD

**Issue by leppaott** [prisma/prisma#19997]
### Bug description

After updating from `4.13.0` to `4.16.1` there seems to be still more type issues:
``` public static findAll(
    select?: Prisma.PrismaLocationTypeSelect
  ): PrismaPromise<Partial<PrismaLocationTypeWithRelations>[]> {
    return prismaClient.prismaLocationType.findMany({ select });
  }
```
breaks with `Type 'undefined' is not assignable to type 'Partial<GetResult<{ uuid: string; name: string;.....`

As the type is following I expect that to work.
`select?: PrismaLocationTypeSelect<ExtArgs> | null`

`select: select ?? {},`

Seems to pass compilation but it can't be empty at runtime. `select ?? null` doesn't work.

Removing that `Partial` still fails `Type 'undefined' is not assignable to type 'GetResult<{ uuid: string;`

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by leppaott** [comment]
Even without directly using that type we have issues:

```
  const select = fields
    ? {
        uuid: true
      }
    : undefined;

  if (sortDirectionByTime) {
    return prismaClient.prismaLocation.findMany({
      select: select,
 ```

**Comment by SevInf** [comment]
Can confirm the bug. Workaround for now, as you mentioned it is providing default value of `{}` to `Select`

**Comment by Jolg42** [comment]
:shipit: Note: This fix is now available as [`4.16.2`](https://github.com/prisma/prisma/releases/tag/4.16.2)
